### PR TITLE
Update `ProofRule::CHAIN_RESOLUTION` to take two lists

### PR DESCRIPTION
**Pull request by ajreynol**
This updates chain resolution and the ALF signature to be in sync.

### CODE REVIEW

**Comment by ajreynol** [comment]
I agree, but these can be cleaned up by making `MACRO_RESOLUTION` take the same arguments as `CHAIN_RESOLUTION`.  This is a deeper refactoring that perhaps @HanielB can address.

File: src/proof/alethe/alethe_post_processor.cpp [hunk 1]
```diff
@@ -436,15 +436,28 @@ bool AletheProofPostprocessCallback::update(Node res,
     case ProofRule::CHAIN_RESOLUTION:
     {
       std::vector<Node> newArgs;
+      std::vector<Node> cargs;
+      if (id == ProofRule::CHAIN_RESOLUTION)
```
Comment (HanielB) [inline src/proof/alethe/alethe_post_processor.cpp:440]: this test does not make sense since this is within a case where id is chain_resolution

**Review by HanielB — CHANGES_REQUESTED** [review]

File: src/proof/alethe/alethe_post_processor.cpp (hunk 1 above)
Comment (ajreynol) [reply src/proof/alethe/alethe_post_processor.cpp:440]: There are 2 cases RESOLUTION and CHAIN_RESOLUTION handled here.

Comment (HanielB) [reply src/proof/alethe/alethe_post_processor.cpp:440]: sorry, github fooled me :)

**Review by HanielB — APPROVED** [review]
I was afraid the changes could impact the complicated function to generate the chain_resolution steps, but looking at them carefully, it looks good to me. 

Also I agree that cleaning this to have less repetition of these conversions would require deeper changes that are better left for the future.

**Pull request merged**